src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -253,31 +253,53 @@ private TempUser createTempUser() {
         tempUser.setSection(user.getSection());
 
         tempUserRepository.save(tempUser);
-        return tempUser;
+    }
+
+    public boolean checkTempUser(UserUpdateDto tempUser) {
+        User user = getAuthenticatedUser();
+
+        return tempUser.getBirthDate().equals(user.getBirthDate()) && tempUser.getCityId().equals(user.getCity().getId()) && tempUser.getCountryId().equals(user.getCountry().getId()) &&
+                tempUser.getPlatoonId().equals(user.getPlatoon().getId()) && tempUser.getRegionId().equals(user.getRegion().getId()) && tempUser.getSectionId().equals(user.getSection().getId()) &&
+                tempUser.getFirstName().equals(user.getFirstName()) && tempUser.getLastName().equals(user.getLastName()) && tempUser.getGender().equals(user.getGender()) &&
+                tempUser.getTelephoneNumber().equals(user.getTelephoneNumber()) && tempUser.getUserAgeGroup().equals(user.getUserAgeGroup()) &&
+                tempUser.getUserRank().equals(user.getUserRank());
     }
 
     public TempUser getTempUserById(Long id) {
         return tempUserRepository.findOne(id);
     }
 
     public void updateTempUser(UserUpdateDto update) {
-        TempUser user = createTempUser();
-
-        user.setUpdateDate(new Date());
-        user.setFirstName(update.getFirstName());
-        user.setLastName(update.getLastName());
-        user.setGender(update.getGender());
-        user.setBirthDate(update.getBirthDate());
-        user.setTelephoneNumber(update.getTelephoneNumber());
-        user.setUserAgeGroup(update.getUserAgeGroup());
-        user.setUserRank(update.getUserRank());
-        user.setCountry(countryRepository.findOne(update.getCountryId()));
-        user.setRegion(regionRepository.findOne(update.getRegionId()));
-        user.setCity(cityRepository.findOne(update.getCityId()));
-        user.setPlatoon(platoonRepository.findOne(update.getPlatoonId()));
-        user.setSection(sectionRepository.findOne(update.getSectionId()));
+        if (tempUserRepository.findByUserId(getAuthenticatedUserId()) == null) {
+            createTempUser();
+            updateTempUsr(update);
+        } else {
+            updateTempUsr(update);
+        }
+    }
```

Comment:
This if - else statement does not looks well. 
We should have only one method saveTempUser() - that will create or update user under the hood.